### PR TITLE
[SCB-2251] make exception processor extendable

### PR DESCRIPTION
**Pull request by wujimin**
Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/SCB) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
 - [ ] Each commit in the pull request should have a meaningful subject line and body.
 - [ ] Format the pull request title like `[SCB-XXX] Fixes bug in ApproximateQuantiles`, where you replace `SCB-XXX` with the appropriate JIRA issue.
 - [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [ ] Run `mvn clean install -Pit` to make sure basic checks pass. A more thorough check will be performed on your pull request automatically.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---

### CODE REVIEW

File: core/src/main/java/org/apache/servicecomb/core/exception/DefaultExceptionProcessor.java [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+  @Override
+  public InvocationException convert(@Nullable Invocation invocation, Throwable throwable, StatusType genericStatus) {
+    Throwable unwrapped = ExceptionFactory.unwrap(throwable);
+    try {
+      return converterCache.computeIfAbsent(unwrapped.getClass(), clazz -> findConverter(unwrapped))
+          .convert(invocation, unwrapped, genericStatus);
+    } catch (Exception e) {
+      LOGGER.error("BUG: ExceptionConverter.convert MUST not throw exception, please fix it.\n"
+              + "original exception :{}"
+              + "converter exception:{}",
```
Comment (liubao68) [inline core/src/main/java/org/apache/servicecomb/core/exception/DefaultExceptionProcessor.java:126]: add blank `converter exception` -> `converter exception `

File: core/src/main/java/org/apache/servicecomb/core/exception/DefaultExceptionProcessor.java [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+  private final List<ExceptionConverter<Throwable>> converters = SPIServiceUtils
+      .getOrLoadSortedService(ExceptionConverter.class).stream()
+      .map(converter -> (ExceptionConverter<Throwable>) converter)
+      .collect(Collectors.toList());
+
+  @InjectProperty(
+      prefix = "servicecomb",
+      keys = {
+          "invocation.exception.print-stack-trace",
+          "exception.invocation.print-stack-trace"
```
Comment (liubao68) [inline core/src/main/java/org/apache/servicecomb/core/exception/DefaultExceptionProcessor.java:68]: why add two configs?

File: core/src/main/java/org/apache/servicecomb/core/exception/DefaultExceptionProcessor.java (hunk 1 above)
Comment (wujimin) [reply core/src/main/java/org/apache/servicecomb/core/exception/DefaultExceptionProcessor.java:126]: to align main content of original and converter

File: core/src/main/java/org/apache/servicecomb/core/exception/DefaultExceptionProcessor.java (hunk 2 above)
Comment (wujimin) [reply core/src/main/java/org/apache/servicecomb/core/exception/DefaultExceptionProcessor.java:68]: servicecomb.exception.invocation.print-stack-trace is for compatible  

others planed same to 
```
sevicecomb.invocation.timeout.check.enabled
sevicecomb.invocation.timeout.check.strategy
sevicecomb.invocation.exception.print-stack-trace
sevicecomb.invocation.exception.print-rate-limit
```

Comment (wujimin) [reply core/src/main/java/org/apache/servicecomb/core/exception/DefaultExceptionProcessor.java:68]: not released, no need to make compatible

Comment (wujimin) [reply core/src/main/java/org/apache/servicecomb/core/exception/DefaultExceptionProcessor.java]: done

**Review by liubao68 — APPROVED** [review]

**Pull request merged**